students/mattmaeda/session04/trigrams.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+import re
+from random import randint
+
+def convert_to_word_list(book_text):
+    raw_list =  re.sub('[^0-9a-zA-Z]+', ' ', book_text).split(" ")
+    return [word.lower() for word in raw_list if word != ""]
+
+
+def build_word_dict(word_list):
+    word_dict = {}
+    for i in range(len(word_list)-2):
+        pair = tuple(word_list[i:i+2])
+        follower = word_list[i+2]
+        word_dict.setdefault(pair, []).append(follower)
+
+    return word_dict
+
+
+def build_new_text(word_dict):
+    rand_idx = randint(0, len(word_dict.keys()))
```

Comment:
you can get the number of entries in a dict with `len(the_dict)` -- no need for the keys() call.